### PR TITLE
Update of 'How do I ...' page in xcube documentation

### PR DESCRIPTION
**Pull request by konstntokas**
This PR adds further examples to the 'How do I ..' page in xcube documentation.

Checklist:

* [ ] ~Add unit tests and/or doctests in docstrings~
* [ ] ~Add docstrings and API docs for any new/modified user-facing classes and functions~
* [x] New/modified features documented in `docs/source/*`
* [x] Changes documented in `CHANGES.md`
* [ ] GitHub CI passes
* [ ] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

**Comment by konstntokas** [comment]
So far the markdown tables are rendered as horizontally scrollable tables. I hard-coded some line break so that the table is not to wide. Maybe you know something which auto-fits the table and supports some auto line beaks. I could not find anything appropriate quickly.  

Screenshot from 2024-06-21 11-45-24

File: docs/source/howdoi.md [hunk 1]
```diff
@@ -1,20 +1,47 @@
...
+| run the xcube server from the CLI     | [xcube serve --verbose -c config.yml](examples/xcube_serve.html#running-the-server)                                                                                                                  | Detailed documentation on `xcube serve` can be found [here](cli/xcube_serve.html) and [here](webapi.html#web-api-and-server). |
+| open the xcube viewer web-application | [xcube serve --open-viewer](examples/xcube_serve.html#id1)                                                                                                                                           | Detailed documentation on xcube Viewer can be found [here](viewer.html).                                                      |
+| start the xcube viewer via Python API | [viewer = Viewer()](api.html#xcube.webapi.viewer.Viewer)<br/>[viewer.add_dataset(ds)](api.html#xcube.webapi.viewer.Viewer.add_dataset)<br/>[viewer.show()](api.html#xcube.webapi.viewer.Viewer.show) | Detailed documentation on xcube Viewer can be found [here](viewer.html).                                                      |
```
Comment (forman) [inline docs/source/howdoi.md:45]:
```suggestion
| start the xcube viewer via Python API<br/>in Jupyter Notebooks | [viewer = Viewer()](api.html#xcube.webapi.viewer.Viewer)<br/>[viewer.add_dataset(ds)](api.html#xcube.webapi.viewer.Viewer.add_dataset)<br/>[viewer.show()](api.html#xcube.webapi.viewer.Viewer.show) | Detailed documentation on xcube Viewer can be found [here](viewer.html).                                                      |
```

Comment (forman) [inline docs/source/howdoi.md:45]: May be better to refer to the examples Notebook here

File: docs/source/howdoi.md [hunk 2]
```diff
@@ -1,20 +1,47 @@
...
+| open a dataset from a data store             | [store.open_data(data_id, **open_params)](api.html#xcube.core.store.DataStore.open_data)                   | `open_params` can be extracted from the store, shown in the next row.                                                                                                                                     |
+| get the parameters for <br>opening a dataset | [store.get_open_data_params_schema()](api.html#xcube.core.store.DataStore.get_open_data_params_schema)     |                                                                                                                                                                                                           |
+
+## get a data store xcube plug-in
+
+| How do I ...                                                                                             | Use API                                                                                        | Comments                                                                              |
```
Comment (forman) [inline docs/source/howdoi.md:18]: In this section, refer to the example notebooks

File: docs/source/howdoi.md [hunk 3]
```diff
@@ -1,20 +1,47 @@
...
+| get a data store for Copernicus Climate Data Store                                                       | [new_data_store("cds", **data_store_params)](api.html#xcube.core.store.new_data_store)         | Documentation can be found [here](dataaccess.html#copernicus-climate-data-store-cds). |
+| get a data store for Copernicus Marine Service                                                           | [new_data_store("cmems", **data_store_params)](api.html#xcube.core.store.new_data_store)       | Documentation can be found [here](dataaccess.html#copernicus-marine-service-cmems).   |
+| get a data store for data organized by <br>a SpatioTemporal Asset Catalog (STAC) (**under development**) | [new_data_store("stac", **data_store_params)](api.html#xcube.core.store.new_data_store)        |                                                                                       |
+
+
+## process data
+
+| How do I ...                                    | Use API                                                                                          | Comments                                                                                     |
```
Comment (forman) [inline docs/source/howdoi.md:31]: In this section, refer to the example notebooks

**Review by forman — COMMENTED** [review]
Would be good to link our dedicated example notbeooks now that we hava a comment column. But that's optional, may enhance later. You might consider this as approved.

**Review by forman — APPROVED** [review]

**Comment by forman** [comment]
Merging, broken CI is due to numpy 2.0

**Pull request merged**